services/vehicle/src/main/java/com/wedriveu/services/vehicle/boundary/nearest/VehicleFinderVerticle.java

```diff
@@ -76,27 +78,24 @@ private void startVehicleCommunication() {
         client = RabbitMQConfig.getInstance(vertx).getRabbitMQClient();
         client.start(onStartCompleted -> {
                     if (onStartCompleted.succeeded()) {
-                        Log.info(TAG, STARTED);
                         declareExchanges(onDeclareCompleted -> {
                             if (onDeclareCompleted.succeeded()) {
-                                Log.info(TAG, EXCHANGE_DECLARED_LOG);
-                                for (int index = ZERO; index < availableVehicles.size(); index++) {
-                                    publishToConsumer(Constants.VEHICLE_SERVICE_EXCHANGE,
-                                            String.format(Constants.ROUTING_KEY_CAN_DRIVE,
-                                                    availableVehicles.get(index).getCarLicencePlate()),
-                                            index);
-                                }
-                            } else {
-                                Log.error(TAG, onDeclareCompleted.cause().getMessage(), onDeclareCompleted.cause());
+                                publishToMultipleRoutingKeys();
                             }
                         });
-                    } else {
-                        Log.error(TAG, onStartCompleted.cause().getMessage(), onStartCompleted.cause());
                     }
                 }
         );
     }
 
+    private void publishToMultipleRoutingKeys() {
+        for (int index = ZERO; index < availableVehicles.size(); index++) {
```

Comment:
Another advice, i think you know we have java 8 here, and so...`IntStream.range(0, availableVehicles.size()).forEach(index -> {...})`